### PR TITLE
Change tchannel.thrift.register to return the unchanged handler.

### PR DESCRIPTION
**Pull request by abhinav**
This is a backport of #302 to 0.21.

### CODE REVIEW

**Comment by abhinav** [comment]
@blampe @breerly @junchaowu

File: tests/schemes/test_thrift.py [hunk 1]
```diff
@@ -1162,3 +1163,80 @@ def testException(request):
...
+def test_void_with_headers_call_directly(server, ThriftTest):
+
+    @server.thrift.register(ThriftTest)
+    def testVoid(request):
+        assert request.headers == {'req': 'header'}
+        return Response(headers={'resp': 'header'})
+
+    resp = testVoid(Request(headers={'req': 'header'}))
+    assert resp.headers == {'resp': 'header'}
+    assert resp.body is None
```
Comment (breerly) [inline tests/schemes/test_thrift.py]: This is awesome.

File: tests/schemes/test_thrift.py [hunk 2]
```diff
@@ -1162,3 +1163,80 @@ def testException(request):
...
+    def testStruct(request):
+        assert request.headers == {'req': 'header'}
+        assert request.body.thing.string_thing == 'req string'
+
+        return Response(
+            server_ttypes.Xtruct(string_thing="resp string"),
+            headers={'resp': 'header'},
+        )
+
+    resp = testStruct(
```
Comment (breerly) [inline tests/schemes/test_thrift.py]: so stoked on this.

File: tests/schemes/test_thrift.py [hunk 3]
```diff
@@ -1162,3 +1163,80 @@ def testException(request):
...
 
     assert 1 == res.status_code
+
+
+##############################################################################
+# Calling registered functions directly should be equivalent to calling them
+# as if they weren't registered at all
+
+
+def body(**kwargs):
```
Comment (breerly) [inline tests/schemes/test_thrift.py]: guessing we'll have to do work to make the request easier to inject into the endpoint?

**Comment by breerly** [comment]
Pumped. :shipit:

File: tests/schemes/test_thrift.py (hunk 3 above)
Comment (abhinav) [inline tests/schemes/test_thrift.py]: Yes, once we settle on an API to let users construct the args structs for different service methods.

I was thinking something like, but let's discuss that elsewhere.

```python
@tchannel.thrift.register
def doFoo(request):
    # ...

def test_do_foo():
    # ...
    doFoo(Request(body=doFoo.args(1, 2, 3)))
```

**Pull request merged**